src/main/java/com/google/ehub/data/EntertainmentItemDatastore.java

```diff
@@ -12,54 +13,97 @@
 import com.google.ehub.utility.NextIncreasingLexicographicStringUtility;
 import java.util.ArrayList;
 import java.util.List;
-
+import java.util.Optional;
 
 /**
- * Manages EntertainmentItems stored in Datastore.
+ * Singleton class that manages EntertainmentItems stored in Datastore.
  */
 public final class EntertainmentItemDatastore {
-  public static final String ENTERTAINMENT_ITEM_KIND = "entertainmentItem";
-  public static final String TITLE_PROPERTY_KEY = "title";
-  public static final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
-  public static final String DESCRIPTION_PROPERTY_KEY = "description";
-  public static final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+  private final String ENTERTAINMENT_ITEM_KIND = "entertainmentItem";
+  private final String UNIQUE_ID_PROPERTY_KEY = "uniqueID";
+  private final String TITLE_PROPERTY_KEY = "title";
+  private final String LOWERCASE_TITLE_PROPERTY_KEY = "lowercaseTitle";
+  private final String DESCRIPTION_PROPERTY_KEY = "description";
+  private final String IMAGE_URL_PROPERTY_KEY = "imageURL";
+
+  private final DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+
+  private static EntertainmentItemDatastore instance;
+
+  private EntertainmentItemDatastore() {}
+
+  /**
+   * Gives access to the single instance of the class, and creates this instance if it was not
+   * initialized previously.
+   *
+   * @return single instance of the class
+   */
+  public static EntertainmentItemDatastore getInstance() {
+    if (instance == null) {
+      instance = new EntertainmentItemDatastore();
+    }
 
-  private static DatastoreService datastoreService = DatastoreServiceFactory.getDatastoreService();
+    return instance;
+  }
 
   /**
    * Adds an EntertainmentItem to Datastore.
    *
-   * @param title title of the entertainment item
-   * @param description description of the entertainment item
-   * @param imageURL URL to the image of the entertainment item stored in Blobstore
+   * @param entertainmentItem entertainment item to add into Datastore
    */
-  public static void addEntertainmentItemToDatastore(
-      String title, String description, String imageURL) {
+  public void addItemToDatastore(EntertainmentItem entertainmentItem) {
     Entity entertainmentItemEntity = new Entity(ENTERTAINMENT_ITEM_KIND);
 
-    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, title);
-    entertainmentItemEntity.setProperty(LOWERCASE_TITLE_PROPERTY_KEY, title.toLowerCase());
-    entertainmentItemEntity.setProperty(DESCRIPTION_PROPERTY_KEY, description);
-    entertainmentItemEntity.setProperty(IMAGE_URL_PROPERTY_KEY, imageURL);
+    entertainmentItemEntity.setProperty(TITLE_PROPERTY_KEY, entertainmentItem.getTitle());
```

Comment:
shouldn't you be setting the id as a property somewhere here?

Reply:
The ID for the item is not a property, Datastore assigns a random ID when the entity is created. The id can be accessed with Entity.getKey().getId(). I could add it as a property as well so that it resembles the class EntertainmentItem, but not sure if that would be better.

Reply:
understood; I think if Datastore already handles ID separately, there's no need to store it as a property, can you add a comment somewhere about this behavior?